src/components/tx/SignOrExecuteForm/hooks.ts

```diff
@@ -138,3 +140,29 @@ export const useIsExecutionLoop = (): boolean => {
   const { safeAddress } = useSafeInfo()
   return wallet ? sameString(wallet.address, safeAddress) : false
 }
+
+export const useRecommendedNonce = (
+  safeTx?: SafeTransaction,
+  initialNonce?: number,
+):
+  | {
+      nonce: number
+      safeTxGas: number
+    }
+  | undefined => {
+  const { safe } = useSafeInfo()
+
+  const [data] = useAsync(() => {
+    if (!safeTx) return
+    return getRecommendedTxParams(safeTx)
+  }, [safeTx, safe?.txQueuedTag])
```

Comment:
Should we also react to the `txHistoryTag`?

Reply:
No, the nonce is incremented only when a tx is added to the queue.